alinka/widget/containers/main_body/content/application/application_tabs/school_tab/__init__.py

```diff
@@ -205,34 +224,57 @@ def selection_changed(self):
 
     @property
     def is_selected(self) -> bool:
-        selected_indexes = self.table_view.selectionModel().selectedRows()
-        return selected_indexes and len(selected_indexes) == 1
+        """Check if exactly one checkbox is selected"""
+        checked_count = 0
+        for row in range(self.model.rowCount()):
+            name_item = self.model.item(row, 0)
+            if name_item and name_item.checkState() == Qt.Checked:
+                checked_count += 1
+        return checked_count == 1
 
     def get_selected_school(self) -> SchoolDbSchema | None:
-        if not self.is_selected:
-            return None
-
-        selected_row = self.table_view.selectionModel().selectedRows()[0].row()
-        school_id = self.model.item(selected_row, 0).data()
-        return get_school_by_id(school_id)
+        """Get the school data for the checked row"""
+        for row in range(self.model.rowCount()):
+            name_item = self.model.item(row, 0)
+            if name_item and name_item.checkState() == Qt.Checked:
+                school_id = name_item.data()  # School ID is stored in the name column
+                return get_school_by_id(school_id)
+        return None
 
     def populate_school_list(self):
         self.model.setRowCount(0)
         school_type = self.school_type.combobox.currentText()
         schools = get_schools() if not school_type else filter_schools_by_type(school_type)
         for school in schools:
+            # Create name item with checkbox
+            name_item = QStandardItem(school.name)
+            name_item.setCheckable(True)
+            name_item.setCheckState(Qt.Unchecked)
+            name_item.setEditable(False)
+            name_item.setData(school.id)  # Store school ID in the name item
+
             row = [
-                QStandardItem(school.name),
+                name_item,
                 QStandardItem(school.type),
                 QStandardItem(school.address),
                 QStandardItem(school.town),
             ]
-            for item in row:
-                item.setData(school.id)
+
+            # Make all columns except name non-editable
+            for c in row[1:]:
```

Comment:
That seems to be error-prone. It would be better to have table definition with columns and which one are editable. Anyway, not for now.